src/sentry/api/endpoints/project_details.py

```diff
@@ -771,6 +778,36 @@ def delete(self, request: Request, project) -> Response:
 
         return Response(status=204)
 
+    @staticmethod
+    def _is_uniform_sampling_rule(rule):
+        # Currently UI sends top level condition op as always 'and' so asserting this here to
+        # guarantee that it will always have an `inner` key
+        assert rule["condition"]["op"] == "and"
```

Comment:
This creates tight coupling with the frontend logic. To be more robust, wouldn't it be better to detect more variants of the uniform sampling rule here?

Reply:
Yeah I thought about that, and then I checked `RuleCondition` in https://github.com/getsentry/relay/blob/cdc6eaddaa222907ea6efd31657bcf4ec998c29c/relay-sampling/src/lib.rs#L251
and iiuc top level condition always has to be an `and` or an `or`, and afaik we are not supporting `or` so not sure what I should assert on here otherwise

Reply:
I think that the implementation is ok, only the comment is misleading.
The rule **must** be an `and` and this has nothing to do with the UI. 
An `or` with no rules will always evaluate to `False`.